fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsSchedularServiceImpl.java

```diff
@@ -65,23 +86,61 @@ public void postInterestForAccounts() throws JobExecutionException {
         Integer initialSize = 500;
         Integer totalPageSize = 0;
         List<Throwable> errors = new ArrayList<>();
-        do {
-            PageRequest pageRequest = PageRequest.of(page, initialSize);
-            Page<SavingsAccount> savingsAccounts = this.savingsAccountRepository.findByStatus(ACTIVE.getValue(), pageRequest);
-            for (SavingsAccount savingsAccount : savingsAccounts.getContent()) {
-                try {
-                    this.savingAccountAssembler.assignSavingAccountHelpers(savingsAccount);
-                    boolean postInterestAsOn = false;
-                    LocalDate transactionDate = null;
-                    this.savingsAccountWritePlatformService.postInterest(savingsAccount, postInterestAsOn, transactionDate);
-                } catch (Exception e) {
-                    LOG.error("Failed to post interest for Savings with id {}", savingsAccount.getId(), e);
-                    errors.add(e);
-                }
+
+        if (this.globalConfigurationRepository.findOneByName("allow-split-interest-posting-on-savings").isEnabled()) {
```

Comment:
allow-split-interest-posting-on-savings-based-on-office  would be more meaningful?